account_credit_control/wizard/credit_control_communication.py

```diff
@@ -0,0 +1,221 @@
+                                                 group['policy_level_id'],
+                                                 group['currency_id']
+                                                 )
+
+            data['credit_control_line_ids'] = [(6, 0, level_lines.ids)]
+            data['partner_id'] = group['partner_id']
+            data['current_policy_level'] = group['policy_level_id']
+            data['currency_id'] = group['currency_id'] or company_currency.id
+            comm = self.create(data)
+            comms += comm
+        return comms
+
+    @api.multi
+    @api.returns('mail.mail')
+    def _generate_emails(self):
+        """ Generate email message using template related to level """
+        emails = self.env['mail.mail']
+        required_fields = ['subject',
+                           'body_html',
+                           'email_from',
+                           'email_to']
+        for comm in self:
+            template = comm.current_policy_level.email_template_id
+            email_values = template.generate_email(comm.id)
+            email_values['message_type'] = 'email'
+            # model is Transient record (self) removed periodically so no point
+            # of storing res_id
+            email_values.pop('model', None)
+            email_values.pop('res_id', None)
+            email = emails.create(email_values)
+
+            state = 'sent'
+            # The mail will not be send, however it will be in the pool, in an
+            # error state. So we create it, link it with
+            # the credit control line
+            # and put this latter in a `email_error` state we not that we have
+            # a problem with the email
+            if not all(email_values.get(field) for field in required_fields):
+                state = 'email_error'
+
+            comm.credit_control_line_ids.write({'mail_message_id': email.id,
+                                                'state': state})
+            attachments = self.env['ir.attachment']
+            for att in email_values.get('attachments', []):
+                attach_fname = att[0]
+                attach_datas = att[1]
+                data_attach = {
+                    'name': attach_fname,
+                    'datas': attach_datas,
+                    'datas_fname': attach_fname,
+                    'res_model': 'mail.mail',
+                    'res_id': email.id,
+                    'type': 'binary',
+                }
+                attachments |= attachments.create(data_attach)
+            email.write({'attachment_ids': [(6, 0, attachments.ids)]})
+            emails += email
+        return emails
+
+    @api.multi
+    def _generate_report(self):
+        """ Will generate a report by inserting mako template
+        of related policy template
+
+        """
+        report_name = 'account_credit_control.report_credit_control_summary'
+        return self.env['report'].get_pdf(self, report_name)
```

Comment:
This part is still to be ported. Method `get_pdf()` and model `report` are not anymore in Odoo 11.0